workflow/controller/operator.go

```diff
@@ -797,29 +797,36 @@ func assessNodeStatus(pod *apiv1.Pod, node *wfv1.NodeStatus) *wfv1.NodeStatus {
 		}
 		newDaemonStatus = pointer.BoolPtr(false)
 	case apiv1.PodRunning:
-		newPhase = wfv1.NodeRunning
-		tmplStr, ok := pod.Annotations[common.AnnotationKeyTemplate]
-		if !ok {
-			log.Warnf("%s missing template annotation", pod.ObjectMeta.Name)
-			return nil
-		}
-		var tmpl wfv1.Template
-		err := json.Unmarshal([]byte(tmplStr), &tmpl)
-		if err != nil {
-			log.Warnf("%s template annotation unreadable: %v", pod.ObjectMeta.Name, err)
-			return nil
-		}
-		if tmpl.Daemon != nil && *tmpl.Daemon {
-			// pod is running and template is marked daemon. check if everything is ready
-			for _, ctrStatus := range pod.Status.ContainerStatuses {
-				if !ctrStatus.Ready {
-					return nil
+		if pod.DeletionTimestamp != nil {
+			// pod is being terminated
+			newPhase = wfv1.NodeError
+			message = "pod termination"
+			log.Info(message)
```

Comment:
Logging should really be structured, so we can create metrics and dashboard, could you make sure this uses `log.WithFields(log.Fields{}).Info("")` to log the workflow name please?

Reply:
Don't need to, it has been set as default in https://github.com/argoproj/argo/blob/master/workflow/controller/operator.go#L110-L112

Reply:
My bad, ignoring previous comment. I'll make change.

Reply:
I simply removed the redundant log, since there are 2 other logs for "node update" and "message update" around, and the "message" is already included in them. 

The best way to use `log` in this function is to turn it to a struct function so that the `log` in the struct can be used, which already has `WithFields()`.